lib/src/models/submission_impl.dart

```diff
@@ -382,3 +387,179 @@ class SubmissionRef extends UserContent {
     return submission;
   }
 }
+
+String _commentSortTypeToString(CommentSortType t) {
+  switch (t) {
+    case CommentSortType.confidence:
+      return 'confidence';
+    case CommentSortType.top:
+      return 'top';
+    case CommentSortType.newest:
+      return 'new';
+    case CommentSortType.controversial:
+      return 'controversial';
+    case CommentSortType.old:
+      return 'old';
+    case CommentSortType.random:
+      return 'random';
+    case CommentSortType.qa:
+      return 'qa';
+    case CommentSortType.blank:
+      return 'blank';
+    default:
+      throw new DRAWInternalError('CommentSortType: $t is not supported.');
+  }
+}
+
+enum CommentSortType {
+  confidence,
+  top,
+  newest,
+  controversial,
+  old,
+  random,
+  qa,
+  blank
+}
+
+class SubmissionModeration extends Object with UserContentModerationMixin {
+  static final RegExp _subModRegExp = new RegExp(r'{subreddit}');
+
+  Submission get content => _content;
+  final Submission _content;
+
+  SubmissionModeration._(this._content);
+
+  /// Enables contest mode for the [Comment]s in this [Submission].
+  ///
+  /// If `state` is true (default), contest mode is enabled for this
+  /// [Submission].
+  ///
+  /// Contest mode have the following effects:
+  ///     * The comment thread will default to being sorted randomly.
+  ///     * Replies to top-level comments will be hidden behind
+  ///       "[show replies]" buttons.
+  ///     * Scores will be hidden from non-moderators.
+  ///     * Scores accessed through the API (mobile apps, bots) will be
+  ///       obscured to "1" for non-moderators.
+  Future contestMode({bool state = true}) async => _content.reddit.post(
+      apiPath['contest_mode'],
+      {
+        'id': _content.fullname,
+        'state': state.toString(),
+      },
+      discardResponse: true);
+
+  /// Sets the flair for the [Submission].
+  ///
+  /// `text` is the flair text to be associated with the [Submission].
+  /// `cssClass` is the CSS class to associate with the flair HTML.
+  ///
+  /// This method can only be used by an authenticated user who has moderation
+  /// rights for this [Submission].
+  Future flair({String text = '', String cssClass = ''}) async {
+    final data = {
+      'css_class': cssClass,
+      'link': _content.fullname,
+      'text': text,
+    };
+    var subreddit = _content.subreddit;
+    if (subreddit is! Subreddit) {
+      subreddit = await subreddit.populate();
+    }
+    final url = apiPath['flair']
+        .replaceAll(_subModRegExp, (subreddit as Subreddit).displayName);
```

Comment:
Why are you being forced to cast here? 

```dart
(subreddit as Subreddit).displayName
```

Guessing subreddit could either be of type `[Subreddit]` or `[SubredditRef]`, but both provide the **same** getter for the _displayName_ field.